hack/devtools/msi.sh

```diff
@@ -1,7 +1,17 @@
 #!/bin/bash
 
-# This script creates a mock MSI object to use for local development
-# We use a service principal and certificate as the mock object
+# This script creates a mock cluster MSI object and platform identities to use for local development
+# We use a service principal and certificate as the mock cluster MSI
+
+if [[ -z "$AZURE_SUBSCRIPTION_ID" ]]; then
```

Comment:
Couple small nits that are likely best solved by turning on `shellcheck` - this file also is missing the error preamble, and it's best to wrap variable names in braces and handle the missing case, e.g. `"${AZURE_SUBSCRIPTION_ID:-}"`

Reply:
Thanks for pointing me to shellcheck - I turned it on in my IDE!

I found some other linting errors through shell check - pushed changes for those as well.